envoy/network/listen_socket.h

```diff
@@ -63,11 +63,21 @@ class ConnectionSocket : public virtual Socket {
    */
   virtual void setJA3Hash(absl::string_view ja3_hash) PURE;
 
+  /**
+   * @param ja4Hash Connection ja3 fingerprint hash of the downstream connection.
+   */
+  virtual void setJA4Hash(absl::string_view ja4_hash) PURE;
```

Comment:
Comment for future cleanup (this PR is just following the pattern already present):

Instead of adding named accessors for everything, maybe we should just have a key/value store for the various metadata that can be set here, so we don't have to touch several interfaces just to pass yet another piece of metadata through.

Reply:
Sure, I can open up a PR next :)